### PR TITLE
Far grab non-dynamic entities

### PR DESCRIPTION
**Pull request by druiz17**
Allow the ability to far-grab non-dynamic entities.

### CODE REVIEW

**Comment by druiz17** [comment]
Test plan
- find or create entities that are grabbable but not dynamic
- you should be able to far grab that entity

**Comment by YuppyPlays** [comment]
Awaiting build for QA.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4004/HighFidelity-Beta-PR11560-d0eb6a3ad090694a2d2735fbaf6e08816d501180.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4004/HighFidelity-Beta-PR11560-d0eb6a3ad090694a2d2735fbaf6e08816d501180.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by druiz17** [comment]
@YuppyPlays - just made some changes, please wait for new build. Updated the test plan.

File: scripts/system/controllers/controllerModules/farActionGrabEntity.js [hunk 1]
```diff
@@ -344,6 +345,13 @@ Script.include("/~/system/libraries/controllers.js");
             var args = [this.hand === RIGHT_HAND ? "right" : "left", MyAvatar.sessionUUID];
             Entities.callEntityMethod(this.grabbedThingID, "releaseGrab", args);
 
+            if (this.madeDynamic) {
+                var props = Entities.getEntityProperties(this.grabbedThingID, ["dynamic", "localVelocity"]);
+                props.dynamic = false;
+                props.localVelocity = {x: 0, y: 0, z: 0};
+                Entities.editEntity(this.grabbedThingID, props);
```
Comment (sethalves) [inline scripts/system/controllers/controllerModules/farActionGrabEntity.js]: I think you don't need to getEntityProperties here, just do
``` var props = {}; ```

File: scripts/system/controllers/grab.js [hunk 2]
```diff
@@ -416,6 +425,14 @@ Grabber.prototype.releaseEvent = function(event) {
         if (this.actionID) {
             Entities.deleteAction(this.entityID, this.actionID);
         }
+
+        if (this.madeDynamic) {
+            var entityProps = Entities.getEntityProperties(this.entityID, ["dynamic", "localVelocity"]);
```
Comment (sethalves) [inline scripts/system/controllers/grab.js]: same here -- editEntity wont change properties that don't have keys in the associative array.

**Comment by YuppyPlays** [comment]
Testing complete:

find or create entities that are grabbable but not dynamic **Check**
you should be able to far grab that entity **Check**

All **Passes** here, though nearly gave myself a heart attack by accidentally flipping the sandbox floor upside down with the Create tool, flinging me up in VR :joy:

**Comment by YuppyPlays** [comment]
@druiz17 

Ah, apologies, I will await next build for testing.

**Comment by sethalves** [comment]
https://highfidelity.fogbugz.com/f/cases/2611/Cannot-far-grab-grabbable-objects-if-not-dynamic

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4008/HighFidelity-Beta-PR11560-013e15c9a2cac45f6256d4a1c101bee396bf696b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4008/HighFidelity-Beta-PR11560-013e15c9a2cac45f6256d4a1c101bee396bf696b.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

find or create entities that are grabbable but not dynamic **Check**
you should be able to far grab that entity **Check**
perform the same test in Desktop mode with mouse **Check**

All looks good here **Pass**

**Pull request merged**